### PR TITLE
Add TestBackend for testing

### LINKED ISSUE #67: No (obvious) way to test messages
As far as I can see, the only way to test that a message has been sent is through mocking.

Would you accept a patch that added a backend that looked like this?

```
class TestBackend(Backend):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.reset_messages()

    def send(self, url, data, **kwargs):
        self.messages.append(data)

    def reset_messages(self):
        self.messages = []

    def retrieve_messages(self):
        messages = self.messages
        self.reset_messages()
        return messages
```

This could be used in more-or-less the same way as Django's `core.mail.backends.locmem.EmailBackend`, with the proviso that you need to manually call `.reset_messages()` on the backend before testing that something has sent a message.

### PR DESCRIPTION
**Pull request by inglesp**
Fixes #67.

Other backends are not tested, so I haven't added tests here either.

Thanks for a nice little tool -- we're using it for the PyCon UK ticketing system.

### CODE REVIEW

File: django_slack/__init__.py [hunk 1]
```diff
@@ -288,6 +288,23 @@
...
 This allows you to, for example, use the Requests library to whilst sending
 your message asynchronously via Celery.
+
+Testing
+-------
+
+You can set ``SLACK_BACKEND = django_slack.backends.TestBackend`` when testing
+to store messages to inspect them later::
+
+    from django_slack.utils import get_backend as get_slack_backend
```
Comment (lamby) [inline django_slack/__init__.py]: Why ``as`` here? :)  Seems to be a distraction in the docs, although I can see why one might do this in practice.

File: django_slack/backends.py [hunk 2]
```diff
@@ -71,5 +71,22 @@ def send(self, *args, **kwargs):
         self._send.delay(*args, **kwargs)
 
 
+class TestBackend(Backend):
```
Comment (lamby) [inline django_slack/backends.py]: Adding a quick docstring along the lines of "this is for testing, use retrieve message" would make this perfect.

**Comment by inglesp** [comment]
Thanks -- I think I've addressed your comments.

**Comment by lamby** [comment]
Uploaded to PyPI as ``5.9.0`` … many thanks :)

**Pull request merged**